### DISCUSSION TITLE
STM32F103xx / LPC55S69 Drag n Drop ?

### DISCUSSION THREAD

**Discussion by rafatuski** [ARMmbed/DAPLink#817] [Q&A]
Hi,

Is the Drag and Drop feature available for the STM32F103 or LPC55S69  ?
or is it Just for some chosen IC like LPC11U35 ?

I can flash STM32F103xx or LPC55s69, but flashing the Interface don't give me Drag n Drop and do not mount as a Volume or / drive on both. But both can be used with openOCD.

Best Regards

Rafatuski

**Comment by mbrossard** [comment]
If you look in `projects.yaml`, you will find the definitions for `stm32f103xb_if`
```
    stm32f103xb_if:
        - *module_if
        - *module_hic_stm32f103xb
        - records/family/all_family.yaml
        - records/usb/usb-bulk.yaml
```
  ... and `stm32f103xb_stm32f411re_if`
```
    stm32f103xb_stm32f411re_if:
        - *module_if
        - *module_hic_stm32f103xb
        - records/board/stm32f411re.yaml
```

The difference is that the first project is a generic interface firmware, it does have a little bit of code to detect some targets and apply some reset logic. Whereas the second projects expect an STM32F411RE target and embeds the flash algorithms needed to do drag & drop flashing. We are working (very slowly) towards a future where one could easily assemble a bit of the firmware for the interface and another for the support of a target, as an intermediate step before (possibly) making that second part dynamically possible.

**Reply by rafatuski** [reply]
Dear mbrossard,

thank you for the answer

Best Regards

**Comment by elfmimi** [comment]
Even for LPC11U35, regarding its popularity it's a bit surprising that, there are no such thing as lpc11u35_lpc11u35_if out of the box.
Here is [someone's note](https://qiita.com/toyowata/items/a57ee1dfa884b29c298d) on how to do that. you can grab the prebuilt binary from his dropbox.

[stm32f103xb_stm32f103rb_if](https://github.com/ARMmbed/DAPLink/blob/408a2af71fc476058a7395eef4bf9ba42fcaf93b/projects.yaml#L571-L574) is a thing in DAPLink. but this one only supports the variant with 128KB of flash memory as its target.
If you want to use it with say STM32F103RC, for example, and if you want to utilize all of its 256KB flash, you need to do some tweaking.

As for LPC55S69 it is supported in DAPLink as both interface and target. So something like lpc55s69_lpc55s69_if should be doable rather easily. but someone has to try and test it.

Like **mbrossard** explained, what we need here is dynamic recognition and switching of flash-algos or at least enhancement to flash-algos so to make it capable of dealing with variants with different amount of flash memory. Which is not ready at current moment.
